stratosphere/pm/source/boot2/boot2_api.cpp

```diff
@@ -0,0 +1,294 @@
+            ncm::TitleId{TitleId_Usb},         /* usb */
+            ncm::TitleId{TitleId_Tma},         /* tma */
+            ncm::TitleId{TitleId_Am},          /* am */
+            ncm::TitleId{TitleId_NvServices},  /* nvservices */
+            ncm::TitleId{TitleId_NvnFlinger},  /* nvnflinger */
+            ncm::TitleId{TitleId_Vi},          /* vi */
+            ncm::TitleId{TitleId_Ns},          /* ns */
+            ncm::TitleId{TitleId_LogManager},  /* lm */
+            ncm::TitleId{TitleId_Ppc},         /* ppc */
+            ncm::TitleId{TitleId_Ptm},         /* ptm */
+            ncm::TitleId{TitleId_Hid},         /* hid */
+            ncm::TitleId{TitleId_Audio},       /* audio */
+            ncm::TitleId{TitleId_Lbl},         /* lbl */
+            ncm::TitleId{TitleId_Wlan},        /* wlan */
+            ncm::TitleId{TitleId_Bluetooth},   /* bluetooth */
+            ncm::TitleId{TitleId_BsdSockets},  /* bsdsockets */
+            ncm::TitleId{TitleId_Nifm},        /* nifm */
+            ncm::TitleId{TitleId_Ldn},         /* ldn */
+            ncm::TitleId{TitleId_Account},     /* account */
+            ncm::TitleId{TitleId_Nfc},         /* nfc */
+            ncm::TitleId{TitleId_JpegDec},     /* jpegdec */
+            ncm::TitleId{TitleId_CapSrv},      /* capsrv */
+            ncm::TitleId{TitleId_Ssl},         /* ssl */
+            ncm::TitleId{TitleId_Nim},         /* nim */
+            ncm::TitleId{TitleId_Erpt},        /* erpt */
+            ncm::TitleId{TitleId_Es},          /* es */
+            ncm::TitleId{TitleId_Pctl},        /* pctl */
+            ncm::TitleId{TitleId_Btm},         /* btm */
+            ncm::TitleId{TitleId_Glue},        /* glue */
+         /* ncm::TitleId{TitleId_Eclct}, */    /* eclct */      /* Skip launching error collection in Atmosphere to lessen telemetry. */
+            ncm::TitleId{TitleId_Fatal},       /* fatal */
+            ncm::TitleId{TitleId_Ro},          /* ro */
+            ncm::TitleId{TitleId_Profiler},    /* profiler */
+            ncm::TitleId{TitleId_Sdb},         /* sdb */
+            ncm::TitleId{TitleId_Migration},   /* migration */
+            ncm::TitleId{TitleId_Grc},         /* grc */
+            ncm::TitleId{TitleId_Olsc},        /* olsc */
+        };
+        constexpr size_t NumAdditionalMaintenanceLaunchPrograms = sizeof(AdditionalMaintenanceLaunchPrograms) / sizeof(AdditionalMaintenanceLaunchPrograms[0]);
+
+        /* Helpers. */
+        inline bool IsHexadecimal(const char *str) {
+            while (*str) {
+                if (!std::isxdigit(static_cast<unsigned char>(*(str++)))) {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        void LaunchTitle(u64 *out_process_id, const ncm::TitleLocation &loc, u32 launch_flags) {
+            u64 process_id = 0;
+
+            /* Don't launch a title twice during boot2. */
```

Comment:
How does this happen? `boot2.flag`? Does Nintendo check this?

Reply:
`boot2.flag` checks this... https://github.com/Atmosphere-NX/Atmosphere/pull/606/files#diff-0225a52181f3642dc4fc0c7845de571dR234

Reply:
Oh, so it does. Nice, I'll remove the check then.